SCClassLibrary/Common/Files/SoundFile.sc

```diff
@@ -420,24 +420,23 @@ SoundFile {
 						Out.ar(out, sig * env * amp)
 					}).add;
 					~instrument = defname;
-					condition = Condition.new;
-					server.sync(condition);
+					server.sync;
 				};
 				ev.synth;	// set up as a synth event (see Event)
 				~bufnum =  server.bufferAllocator.alloc(1);
-				~bufferSize = 0x10000;
+				~bufferSize = ~bufferSize ? 0x10000;
 				~firstFrame = ~firstFrame ? 0;
 				~lastFrame = ~lastFrame ? numFrames;
-				~sustain = (~lastFrame - ~firstFrame)/(sampleRate ?? {server.options.sampleRate ? 44100});
+				~sustainTime = (~lastFrame - ~firstFrame)/(sampleRate ?? {server.sampleRate ? 44100});
 				~close = { | ev |
 						server.bufferAllocator.free(ev[\bufnum]);
 						server.sendBundle(server.latency, ["/b_close", ev[\bufnum]],
 							["/b_free", ev[\bufnum] ]  )
 				};
-				~setwatchers = { |ev|
+				if (closeWhenDone) {
 					OSCFunc({
-						server.sendBundle(server.latency, ["/b_close", ev[\bufnum]],
-							["/b_read", ev[\bufnum], path, ev[\firstFrame], ev[\bufferSize], 0, 1]);
+						ev.close;
+						this.close;
```

Comment:
if someone adds a method extension:
```supercollider
+ Object {
    close { ... do something ... }
}
```

the `ev.close` will break.

better write:

```ev[\close].value(ev)```